pydantic/_internal/_generate_schema.py

```diff
@@ -1248,7 +1248,14 @@ def _literal_schema(self, literal_type: Any) -> CoreSchema:
         """Generate schema for a Literal."""
         expected = _typing_extra.all_literal_values(literal_type)
         assert expected, f'literal "expected" cannot be empty, obj={literal_type}'
-        return core_schema.literal_schema(expected)
+        schema = core_schema.literal_schema(expected)
+
+        if self._config_wrapper.use_enum_values:
```

Comment:
Let's add one more check, where we only apply this change if there's one or more `Enum`s in `expected.

Reply:
Done!